stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/querychannels/DefaultChatEventHandler.kt

```diff
@@ -47,12 +48,16 @@ internal class DefaultChatEventHandler(private val client: ChatClient, private v
     override fun handleNotificationAddedToChannelEvent(
         event: NotificationAddedToChannelEvent,
         filter: FilterObject,
-    ): EventHandlingResult = fireRequestIfChannelIsAbsent(event.channel, filter)
+    ): EventHandlingResult {
+        return handleMemberUpdate(event, event.cid, filter)
+    }
 
     override fun handleChannelUpdatedByUserEvent(
         event: ChannelUpdatedByUserEvent,
         filter: FilterObject,
-    ): EventHandlingResult = EventHandlingResult.Skip
+    ): EventHandlingResult {
+        return handleMemberUpdate(event, event.cid, filter)
```

Comment:
We can't do this. `ChannelUpdatedByUserEvent` is often update and this logic isn't generic for all possible use-cases (like messaging and live stream. messaging requires a user being a member of channel and live stream doesn't). 
If we do such requests always out-of-box we will throttle our BE and customers reach their rate limit too soon. We can suggest such change to particular customers that works fine in their case. They can inject their own `ChatEventHandler` into `ChannelListViewModel`. We have docs about it [here](https://getstream.io/chat/docs/sdk/android/ui/components/channel-list/#chateventhandler)

Reply:
I took a look at the code and the update wouldn't happen many times, only once. The when case: 
```
when {
                filterPassed && !hasChannel -> EventHandlingResult.Add(channel)

                !filterPassed && hasChannel -> EventHandlingResult.Remove(cid)

                else -> EventHandlingResult.Skip
            }
```
Will once go to the `else` condition because the channel will be updated, so it won't ask to add or remove a user without need.

Reply:
It is filter that does many queries, not the add/remove. Ok, let's add a skip in this event.